src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -464,6 +473,15 @@ export class ExerciseService {
         }
     }
 
+    private setBuildPlanUrlToParticipations(exercise: Exercise | undefined | null) {
+        if (exercise?.type === ExerciseType.PROGRAMMING) {
```

Comment:
Wouldn't it be enough to do this if the publication of the buildplan is activated for that exercise?